File: mxpack-core/src/test/java/org/odenix/mxpack/core/example/ReadArray.java
/*
 * Copyright 2024 the MxPack project authors
 * SPDX-License-Identifier: Apache-2.0
 */
package org.odenix.mxpack.core.example;

import net.jqwik.api.ForAll;
import org.odenix.mxpack.core.MessageReader;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

class ReadArray extends Example {
  // -8<- [start:snippet]
  List<String> read(MessageReader reader) throws IOException {
    var size = reader.readArrayHeader();
    var list = new ArrayList<String>(size);
    for (int i = 0; i < size; i++) {
      var element = reader.readString();
      list.add(element);
    }
    return list;
  }
  // -8<- [end:snippet]

  @net.jqwik.api.Example
  void test(@ForAll List<String> list) throws IOException {
    new WriteArray().test(list);
  }
}

File: mxpack-kotlin/src/test/kotlin/org/odenix/mxpack/kotlin/example/ReadArray.kt
/*
 * Copyright 2024 the MxPack project authors
 * SPDX-License-Identifier: Apache-2.0
 */
package org.odenix.mxpack.kotlin.example

import org.odenix.mxpack.core.MessageReader
import net.jqwik.api.ForAll

class ReadArray : Example() {
  // -8<- [start:snippet]
  fun read(reader: MessageReader): List<String> {
    val size = reader.readArrayHeader()
    return buildList(size) {
      repeat(size) {
        val element = reader.readString()
        add(element)
      }
    }
  }
  // -8<- [end:snippet]

  @net.jqwik.api.Example
  fun test(@ForAll list: List<String>) {
    WriteArray().test(list)
  }
}
